mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -350,7 +345,6 @@ private class LeastSquaresAggregator(
         }
       }
       lossSum += diff * diff / 2.0
-      diffSum += diff
```

Comment:
When I started to implement this code, I thought diffSum will not be zero. @mengxr caught that I don't add the correction terms back to the gradient correctly, but I still get the right models compared with R that made me go through the math again, and I found diffSum is always zero theoretically. Here is the proof added in the comment
https://github.com/apache/spark/commit/15995c883aa248235fdebf0cbeeaa3ef12c97e9c

Array type is obvious since it's Array.ofDim. I did consider to remove the Long type, but in the end, I worry about someone can mistakenly remove L from 0L, so I decided to keep it.